File: sdks/node/test/usm.test.ts
import { describe, it, expect } from 'vitest';
import { USM } from '../src/usm';

describe('USM', () => {
  it('should load secrets', async () => {
    // This is a placeholder test
    // In a real test, you would mock the file system
    const usm = await USM.load('./test/fixtures/.secrets.yml');
    expect(usm).toBeInstanceOf(USM);
  });

  it('should get a secret', async () => {
    // This is a placeholder test
    const usm = await USM.load('./test/fixtures/.secrets.yml');
    const value = await usm.get('TEST_KEY');
    expect(value).toBe('decrypted_value_for_TEST_KEY');
  });
});

File: sdks/go/usm_test.go
package usm

import (
	"testing"
)

func TestLoad(t *testing.T) {
	// This is a placeholder test
	// In a real test, you would mock the file system
	usm, err := Load("./test/fixtures/.secrets.yml")
	if err != nil {
		t.Fatalf("Failed to load secrets file: %v", err)
	}

	if usm == nil {
		t.Error("Expected USM instance, got nil")
	}
}

func TestGet(t *testing.T) {
	// This is a placeholder test
	usm, err := Load("./test/fixtures/.secrets.yml")
	if err != nil {
		t.Fatalf("Failed to load secrets file: %v", err)
	}

	value, err := usm.Get("TEST_KEY")
	if err != nil {
		t.Fatalf("Failed to get secret: %v", err)
	}

	if value != "decrypted_value_for_TEST_KEY" {
		t.Errorf("Expected 'decrypted_value_for_TEST_KEY', got '%s'", value)
	}
}